core/trino-main/src/main/java/io/trino/execution/scheduler/UniformNodeSelector.java

```diff
@@ -171,28 +172,28 @@ public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTas
             }
 
             InternalNode chosenNode = null;
-            int min = Integer.MAX_VALUE;
+            long minWeight = Long.MAX_VALUE;
 
             for (InternalNode node : candidateNodes) {
-                int totalSplitCount = assignmentStats.getTotalSplitCount(node);
-                if (totalSplitCount < min && totalSplitCount < maxSplitsPerNode && assignmentStats.getUnacknowledgedSplitCountForStage(node) < maxUnacknowledgedSplitsPerTask) {
+                long totalSplitsWeight = assignmentStats.getTotalSplitsWeight(node);
+                if (totalSplitsWeight < minWeight && totalSplitsWeight < maxSplitsWeightPerNode && assignmentStats.getUnacknowledgedSplitCountForStage(node) < maxUnacknowledgedSplitsPerTask) {
                     chosenNode = node;
-                    min = totalSplitCount;
+                    minWeight = totalSplitsWeight;
                 }
             }
             if (chosenNode == null) {
-                // min is guaranteed to be MAX_VALUE at this line
+                // minWeight is guaranteed to be MAX_VALUE at this line
                 for (InternalNode node : candidateNodes) {
-                    int totalSplitCount = assignmentStats.getQueuedSplitCountForStage(node);
-                    if (totalSplitCount < min && totalSplitCount < maxPendingSplitsPerTask && assignmentStats.getUnacknowledgedSplitCountForStage(node) < maxUnacknowledgedSplitsPerTask) {
+                    long queuedWeight = assignmentStats.getQueuedSplitsWeightForStage(node);
```

Comment:
The `ForStage` name feels confusing. From what I understand it returns a weight of splits assigned to a node for this specific stage, that effectively means the total weight of splits assigned to a task on that specific node.

Reply:
This was just a straight "find: 'count', replace: 'weight'" in method names exercise, so I tried to avoid renaming methods even if they didn't make total sense (yeah, this is more conceptually "ForTask" and not "ForStage").

This isn't _total_ splits on that node because splits that are actively running are not included, and the semantics are a little subtle because splits in any of the following states (for this particular task) are considered "queued" in this context:
- On the worker node, queued but not yet running as of the last task status update
- On the coordinator, not yet sent to the worker but already placed into the `HttpRemoteTask` to be sent in the next update request
- Tentatively assigned to the task during the current split scheduling batch, but not yet placed into the `HttpRemoteTask`